### PR TITLE
[RST-2186] Added a FixedLagSmootherParams struct

### PR DESCRIPTION
**Pull request by svwilliams**
Added a FixedLagSmootherParams struct that is responsible for holding and loading all of the Fixed-lag Smoother settings.

This currently does not include the set of parameters loaded and used by the Optimizer base class. A future PR will make an OptimizerParams base class that loads the common parameters.

### CODE REVIEW

File: fuse_optimizers/src/fixed_lag_smoother.cpp [hunk 1]
```diff
@@ -107,12 +82,15 @@ FixedLagSmoother::FixedLagSmoother(
...
   // Configure a timer to trigger optimizations
   optimize_timer_ = node_handle_.createTimer(
-    optimization_period_,
+    params_.optimization_period,
     &FixedLagSmoother::optimizerTimerCallback,
     this);
 
   // Advertise a service that resets the optimizer to its initial state
-  reset_service_server_ = private_node_handle_.advertiseService("reset", &FixedLagSmoother::resetServiceCallback, this);
+  reset_service_server_ = node_handle_.advertiseService(
```
Comment (svwilliams) [inline fuse_optimizers/src/fixed_lag_smoother.cpp:90]: This switch from `private_node_handle_` to `node_handle_` is intentional. It is in the spirit of locusrobotics/fuse_rl#10

File: fuse_optimizers/include/fuse_optimizers/fixed_lag_smoother_params.h [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+   */
+  ros::Duration optimization_period { 0.1 };
+
+  /**
+   * @brief The topic name of the advertised reset service
+   */
+  std::string reset_service { "reset" };
+
+  /**
+   * @brief The maximum time to wait for motion models to be generated for a received transactions.
```
Comment (matthewbwhitaker) [inline fuse_optimizers/include/fuse_optimizers/fixed_lag_smoother_params.h]: "a received transaction" or "received transactions"?

File: fuse_optimizers/include/fuse_optimizers/fixed_lag_smoother_params.h [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+   *
+   * Transactions are processes sequentially, so no new transactions will be added to the graph while waiting for
```
Comment (matthewbwhitaker) [inline fuse_optimizers/include/fuse_optimizers/fixed_lag_smoother_params.h]: *processed

**Review by matthewbwhitaker — APPROVED** [review]

File: fuse_optimizers/include/fuse_optimizers/fixed_lag_smoother_params.h [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+  /**
+   * @brief Helper function that loads strictly positive floating point values from the parameter server
+   *
+   * @param[in] node_handle - The node handle used to load the parameter
+   * @param[in] parameter_name - The parameter name to load
+   * @param[in] default_value - A default value to use if the provided parameter name does not exist
+   * @return The loaded (or default) value
+   */
+  double getPositiveParam(const ros::NodeHandle& node_handle, const std::string& parameter_name, double default_value)
```
Comment (ayrton04) [inline fuse_optimizers/include/fuse_optimizers/fixed_lag_smoother_params.h:69]: Not for this PR, but if we can't release some of our internal utility methods like this, we should consider adding some of them to a utility header in this package.

**Review by ayrton04 — APPROVED** [review]

**Pull request merged**